app/models/manageiq/providers/azure/cloud_manager/event_catcher/stream.rb

```diff
@@ -29,13 +30,14 @@ def each_batch
   def get_events
     # Grab only events for the last minute if this is the first poll
     filter = @since ? "eventTimestamp ge #{@since}" : "eventTimestamp ge #{startup_interval}"
-    fields = 'authorization,description,eventName,eventTimestamp,resourceGroupName,resourceProviderName,resourceId,resourceType'
+    fields = 'authorization,correlationId,description,eventId,eventName,eventTimestamp,resourceGroupName,resourceProviderName,resourceId,resourceType,submissionTimestamp'
     events = connection.list(:filter => filter, :select => fields, :all => true).sort_by(&:event_timestamp)
 
     # HACK: the Azure Insights API does not support the 'gt' (greater than relational operator)
     # therefore we have to poll from 1 millisecond past the timestamp of the last event to avoid
     # gathering the same event more than once.
-    @since = one_ms_from_last_timestamp(events) unless events.empty?
+    #@since = one_ms_from_last_timestamp(events) unless events.empty?
+    @since = @worker.last_heartbeat unless events.empty?
```

Comment:
I could be wrong (cc @jrafanie) but I think the previous worker record will be destroyed by the time the next worker is started.  Testing this theory out locally.

Reply:
Ah, crud.

Reply:
Yeah so I tried delivering a SIGKILL to my MiqEventHandler:

```
>> MiqEventHandler.first.id
=> 17

$ ps aux | grep MiqEventHandler | grep -v grep | awk '{print $2}' | xargs kill -KILL

... wait for MiqServer to restart it

>> MiqEventHandler.first.id
=> 18
```

Reply:
@djberg96 I saw your comment from https://gitter.im/ManageIQ/manageiq?at=5d72a5c82f93ab5a15190ec3 and assumed by "last time it ran" as when the previous worker was running when the next worker starts up, is that what you want?

Reply:
@agrare That sounds right. How do I do that?

Reply:
Okay, so that's an issue because of [:point_up:](https://github.com/ManageIQ/manageiq-providers-azure/pull/350/files#r321889178)

I don't know of a good way to tell when the previous worker instance was running.

Is that the issue here though?  The worker restarted and the first event that was delivered had a timestamp newer than the next event.

Reply:
@agrare on second thought, maybe that doesn't matter after all, as long as we're getting the latest time that the runner executed. We're tinkering with it now...

Reply:
Why not make `@since` the last time when events were requested then?  I don't think you want last_heartbeat because that is relatively infrequent compared with the event catcher

Reply:
@agrare Yeah, we were thinking of just going with `Time.now - polling_interval`.